### PR TITLE
Remove unused dependencies

### PR DESCRIPTION
**Pull request by davisagli**
This removes dependencies that I believe we are not using -- would appreciate a sanity check on that though.

### CODE REVIEW

File: manage.py [hunk 1]
```diff
@@ -2,10 +2,14 @@
 import os
 import sys
 
-from newrelic import agent
+try:
+    from newrelic import agent
+except ImportError:
+    agent = None
```
Comment (wlonk) [inline manage.py]: Given that you've removed `newrelic`, when do you expect the import to ever succeed?

File: metashare/asgi.py [hunk 2]
```diff
@@ -7,11 +7,15 @@
...
 from channels.routing import get_default_application
-from newrelic import agent
 from sentry_sdk.integrations.asgi import SentryAsgiMiddleware
 
-agent.initialize()
-agent.wrap_web_transaction("django.core.handlers.base", "BaseHandler.get_response")
+try:
+    from newrelic import agent
+except ImportError:
+    pass
```
Comment (wlonk) [inline metashare/asgi.py]: Ditto about `newrelic`.

File: metashare/rq_worker.py [hunk 3]
```diff
@@ -1,20 +1,23 @@
 from django.db import DatabaseError, InterfaceError, connections
-from newrelic import agent
 from rq.job import Job
 from rq.worker import HerokuWorker, Worker
 
+try:
+    from newrelic import agent
+except ImportError:
+    agent = None
```
Comment (wlonk) [inline metashare/rq_worker.py]: And thritto about `newrelic`.

File: metashare/tests/rq_worker.py [hunk 4]
```diff
@@ -4,21 +4,25 @@
...
-        func_name = "func"
-        perform = mock_perform
-
-    wrap_job_as_background_task(Job)
-    Job().perform()
-    mock_perform.assert_called_once()
+try:
+    from newrelic import agent
+except ImportError:
+    agent = None
```
Comment (wlonk) [inline metashare/tests/rq_worker.py]: OK, I'm running out of prefixes. Quitto? 4itto?

**Comment by davisagli** [comment]
Fair question about newrelic. I was keeping the code in case we decide to add it back in, but I may as well take it out entirely since we can recover it from version control.

File: requirements/prod.in [hunk 5]
```diff
@@ -12,21 +10,16 @@ django-hashid-field==3.1.1
...
 furl==2.1.0
 github3-py==1.3.0
 rq-scheduler==0.9.1
 logfmt==0.4
 markdown==3.2.1
-newrelic==5.8.0.136
 psycopg2-binary==2.8.4
-pyOpenSSL==19.1.0
 sentry-sdk==0.14.2
-service_identity==18.1.0
```
Comment (jgerigmeyer) [inline requirements/prod.in]: While building, I now get this error/warning:

```
Step 21/22 : RUN python manage.py collectstatic --noinput
 ---> Running in 9124cc2a254c
:0: UserWarning: You do not have a working installation of the service_identity module: 'No module named 'service_identity''.  Please install it from <https://pypi.python.org/pypi/service_identity> and make sure all of its dependencies are satisfied.  Without the service_identity module, Twisted can perform only rudimentary TLS client hostname verification.  Many valid certificate/hostname mappings may be rejected.
```

I think `channels` requires `daphne`, which requires `twisted[tls]`, which requires `pyopenssl`, `service_identity`, and `idna`. The other two are already installed along with `requests`, but now `service_identity` is missing.

https://github.com/twisted/twisted/blob/trunk/src/twisted/python/_setup.py#L96-L104

Comment (davisagli) [reply requirements/prod.in]: Hmm, okay, I'll revisit. The fact that I was able to run the tests without seeing this warning suggests that our Python tests do not make sure that we can run daphne.

Comment (wlonk) [reply requirements/prod.in]: That's to be expected; the tests are about our logic, not about our ASGI server. But I was able to run it fine to hand-test, as far as I could see, too, which surprises me given that error.

**Comment by davisagli** [comment]
@jgerigmeyer maybe your docker container was built using cached layers so has older versions of something(s)? I just rebuilt mine from the top yesterday after deleting my oddbird/pyjs image to make sure it fetched the newest one.

**Pull request merged**